src/main/java/service/imp/ReaderServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package service.imp;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import service.ReaderService;
+
+public class ReaderServiceImpl implements ReaderService {
+    private List<String> dataFromFile = new ArrayList<>();
```

Comment:
Avoid using class fields for storing data that is specific to a method execution. This can lead to issues in a multi-threaded environment and is not a good practice because it's not thread-safe. Instead, use local variables within the `read` method.